components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -78,6 +80,78 @@ def update_compression_job_metadata(db_cursor, job_id, kv):
     db_cursor.execute(query, values)
 
 
+def _process_fs_input_paths(
+    fs_input_conf: FsInputConfig, paths_to_compress_buffer: PathsToCompressBuffer
+) -> None:
+    """
+    Iterate through all files in fs_input_conf and adds their metadata to the
+    paths_to_compress_buffer.
+    Note: this method skips any files that do not exist.
+    :param fs_input_conf: FS configuration specifying the files to compress.
+    :param paths_to_compress_buffer: PathsToCompressBuffer containing the scheduling information
+    :return: None.
+    """
+
+    for path_idx, path in enumerate(fs_input_conf.paths_to_compress, start=1):
+        path = Path(path)
+
+        try:
+            file, empty_directory = validate_path_and_get_info(CONTAINER_INPUT_LOGS_ROOT_DIR, path)
+        except ValueError as ex:
+            logger.error(str(ex))
+            continue
+
+        if file:
+            paths_to_compress_buffer.add_file(file)
+        elif empty_directory:
+            paths_to_compress_buffer.add_empty_directory(empty_directory)
+
+        if path.is_dir():
+            for internal_path in path.rglob("*"):
+                try:
+                    file, empty_directory = validate_path_and_get_info(
+                        CONTAINER_INPUT_LOGS_ROOT_DIR, internal_path
+                    )
+                except ValueError as ex:
+                    logger.error(str(ex))
+                    continue
+
+                if file:
+                    paths_to_compress_buffer.add_file(file)
+                elif empty_directory:
+                    paths_to_compress_buffer.add_empty_directory(empty_directory)
+
+
+def _process_s3_input(
```

Comment:
Why not use exceptions (instead of `Result`)? Same for `get_s3_object_metadata`.

Reply:
Do you mean use try except outside?

Reply:
Yeah, throw exceptions from the mentioned methods and use try-except outside.

Reply:
Done, but as discussed in our private chat, will need your guidance on some docstrings.